aips/aip-lite-account.md

```diff
@@ -0,0 +1,143 @@
+
+## Alternative solutions
+
+- Modify the current account module to support Account Abstraction with a new resource.
+
+The alternative solution is a patch but not a clean solution. it cannot get rid of the deprecated and unnecessary fields and takes more storage slots so it can never has 0 storage footprint in the future.
+The current Account resource has 7 fields and 5 are absolutely deprecated, 1 is unnecessary for most user and the left 1 will be unnecessary for most people in the future if Aptos has orderless transaction support.
+
+## Specification and Implementation Details
+
+This AIP primarily introduces the following new structs:
+
+In `lite_account.move`:
+```
+    #[resource_group(scope = address)]
+    /// A shared resource group for storing new account resources together in storage.
+    struct LiteAccountGroup {}
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// Resource representing an account object.
+    struct Account has key {
+        sequence_number: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// The native authenticator where the key is used for authenticator verification in native code.
+    struct NativeAuthenticator has key, copy, drop {
+        auth_key: Option<vector<u8>>,
+    }
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// The dispatchable authenticator that defines how to authenticates this account in the specified module.
+    /// An integral part of Account Abstraction.
+    struct DispatchableAuthenticator has key, copy, drop {
+        auth_function: FunctionInfo
+    }
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// Legacy field from deprecated Account module.
+    struct LegacyGUIDCreactionNumber has key {
+        creation_number: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// Legacy field from deprecated Account module.
+    struct LegacyRotationCapabilityOffer has key, drop { for: address }
+
+    #[resource_group_member(group = aptos_framework::lite_account::LiteAccountGroup)]
+    /// Legacy field from deprecated Account module.
+    struct LegacySignerCapabilityOffer has key, drop { for: address }
+```
+
+And other functions to support all the operations on those resources, please refer to the reference implementation for more details.
+
+
+## Reference Implementation
+
+https://github.com/aptos-labs/aptos-core/pull/10780
+
+## Testing (Optional)
+
+All functionality verified with unit tests across aptos-framework, e2e-move-tests and api.
+
+## Risks and Drawbacks
+
+Since this AIP proposes a compatible solution, if the ecosystem still rely on account.move heavily and do not adopt the features of the lite account, the benefits would be greatly deprecated. 
```

Comment:
What about:
1. Performance penalties of this approach?
2. Addressing who pays the gas costs?

Reply:
Please refer to the updated version.
1. discussed.
2. No AA.
3. keyless/passkey are orthogonal.